core/trino-main/src/main/java/io/trino/metadata/ResolvedFunction.java

```diff
@@ -181,6 +189,8 @@ public String toString()
 
     public static class ResolvedFunctionDecoder
     {
+        private static final Compressor COMPRESSOR = new ZstdCompressor();
+        private static final Decompressor DECOMPRESSOR = new ZstdDecompressor();
```

Comment:
These classes are not thread-safe, and as far as I can tell, there's no guarantee that `fromQualifiedName` and friends will be called from a single thread at a time.